addon/-legacy-private/system/relationships/state/belongs-to.js

```diff
@@ -156,11 +169,50 @@ export default class BelongsToRelationship extends Relationship {
   }
 
   findRecord() {
-    if (this.inverseInternalModel) {
-      return this.store._findByInternalModel(this.inverseInternalModel);
+    let promise;
+    let { inverseInternalModel, shouldForceReload } = this;
+
+    if (inverseInternalModel) {
+      if (shouldForceReload && inverseInternalModel.hasRecord) {
+        // reload record, if it is already loaded
+        promise = inverseInternalModel.getRecord().reload();
```

Comment:
if we have a link, shouldn't we go fetch the link instead of reloading the record?